components/schedule-editor/src/ScheduleEditor.svelte

```diff
@@ -248,10 +288,165 @@
+
+    if (fieldSwapRequired) {
+      let updatedFieldOrder = [..._this.custom_fields] as CustomFieldWithId[];
+
+      // Swap the current hovered field with the dragged field
+      let temp = updatedFieldOrder[hoveredFieldIndex];
+      updatedFieldOrder[hoveredFieldIndex] =
+        updatedFieldOrder[draggedFieldIndex];
+      updatedFieldOrder[draggedFieldIndex] = temp;
+
+      // update custom fields to reflect the drag order
+      _this.custom_fields = updatedFieldOrder;
+      customFieldIdOrder = updatedFieldOrder.map((field) => field.id);
+
+      // current dragged index is now what was hovered
+      draggedFieldIndex = updatedFieldOrder.findIndex(
+        (customField) => customField.id === draggedField?.id,
+      );
+    }
+  }
+
+  const throttledDraggedFieldSwap = throttle(swapHoveredWithDraggedFields, 100);
+
+  const setDragRowPosition = (e: MouseEvent) => {
+    const rowHeightOffset = tablerowRect.height / 2;
+    const dragRowPosition = e.pageY;
+
+    // stick the drag preview if cursor is beyond max top and max bottom
+    let previewTopPosition: number | null = null;
+    if (dragRowPosition <= maxDragTop) {
+      previewTopPosition = maxDragTop - rowHeightOffset;
+    } else if (dragRowPosition >= maxDragBottom) {
+      previewTopPosition = maxDragBottom - rowHeightOffset;
+    }
+
+    // set initial position of drag preview
+    dragPlaceholder = {
+      ...dragPlaceholder,
+      left: tablerowRect.x,
+      top: previewTopPosition ?? dragRowPosition - rowHeightOffset,
+    };
+  };
+
+  $: if (customFieldDomRects) {
+    // Set max top and bottom for drag
+    if (customFieldDomRects[0] && !maxDragTop) {
+      maxDragTop =
+        document.documentElement.scrollTop +
+        customFieldDomRects[0].top +
+        tablerowRect.height / 2;
+    }
+    if (customFieldDomRects[customFieldDomRects.length - 1] && !maxDragBottom) {
+      maxDragBottom =
+        document.documentElement.scrollTop +
+        customFieldDomRects[customFieldDomRects.length - 1].top +
+        tablerowRect.height / 2;
+    }
+  }
+
+  function handleCustomFieldDragMove(e: MouseEvent) {
+    if (draggedField) {
+      setDragRowPosition(e);
+      throttledDraggedFieldSwap(e);
+    }
+  }
+
+  function storeRef(node: HTMLElement, params: { id: string }) {
+    customFieldRefs[params.id] = node;
+
+    if (!tablerowRect) {
+      const { parentRect, childRects } = getDomRectsFromParentAndChildren(
+        node,
+        "td",
+      );
+      tablerowRect = parentRect;
+      tablecellRect = childRects;
+
+      // Watches row element if it changes size
+      new ResizeObserver(([observerEntry]) => {
+        const { parentRect, childRects } = getDomRectsFromParentAndChildren(
+          observerEntry.target,
+          "td",
+        );
+        tablerowRect = parentRect;
+        tablecellRect = childRects;
+
+        dragBegins = false; // reset on next start of drag
+      }).observe(node);
```

Comment:
Do we need to stop observing this on destroy, or does it clean itself up?

Reply:
Good catch!